uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/absence-tab/components/absence-tab-entitlements_test.js

```diff
@@ -6,41 +6,59 @@ define([
   'common/angularMocks',
   'leave-absences/shared/config',
   'leave-absences/absence-tab/app',
+  'common/mocks/services/api/contact-mock',
   'common/mocks/services/api/contract-mock',
   'mocks/apis/absence-type-api-mock',
+  'mocks/apis/absence-period-api-mock',
+  'mocks/apis/entitlement-api-mock',
   'leave-absences/shared/modules/shared-settings'
 ], function (moment, angular) {
   'use strict';
 
   describe('absenceTabEntitlements', function () {
-    var $compile, $log, $rootScope, $q, component, controller, Contract,
-      ContractAPIMock, AbsenceType, AbsenceTypeAPIMock;
+    var $compile, $log, $rootScope, $q, component, controller,
+      Contract, ContractAPIMock, Contact, ContactAPIMock,
+      AbsenceType, AbsenceTypeAPIMock, AbsencePeriod, AbsencePeriodAPIMock,
+      Entitlement, EntitlementAPIMock;
 
     beforeEach(module('leave-absences.templates', 'absence-tab',
                       'common.mocks', 'leave-absences.mocks'
     ));
 
-    beforeEach(inject(['api.contract.mock', 'AbsenceTypeAPIMock', function (_ContractAPIMock_, _AbsenceTypeAPIMock_) {
-      ContractAPIMock = _ContractAPIMock_;
-      AbsenceTypeAPIMock = _AbsenceTypeAPIMock_;
-    }]));
+    beforeEach(inject(['api.contract.mock', 'api.contact.mock', 'AbsenceTypeAPIMock', 'AbsencePeriodAPIMock', 'EntitlementAPIMock',
+      function (_ContractAPIMock_, _ContactAPIMock_, _AbsenceTypeAPIMock_, _AbsencePeriodAPIMock_, _EntitlementAPIMock_) {
+        ContractAPIMock = _ContractAPIMock_;
+        ContactAPIMock = _ContactAPIMock_;
+        AbsenceTypeAPIMock = _AbsenceTypeAPIMock_;
+        AbsencePeriodAPIMock = _AbsencePeriodAPIMock_;
+        EntitlementAPIMock = _EntitlementAPIMock_;
+      }]));
 
-    beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _Contract_, _AbsenceType_) {
+    beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _Contract_, _Contact_, _AbsenceType_, _AbsencePeriod_, _Entitlement_) {
       $compile = _$compile_;
       $log = _$log_;
       $rootScope = _$rootScope_;
       $q = _$q_;
 
+      Contact = _Contact_;
       Contract = _Contract_;
       AbsenceType = _AbsenceType_;
+      AbsencePeriod = _AbsencePeriod_;
+      Entitlement = _Entitlement_;
 
       spyOn($log, 'debug');
-      spyOn(AbsenceType, 'all').and.callFake(function () {
-        return $q.resolve(AbsenceTypeAPIMock.all());
-      });
-      spyOn(Contract, 'all').and.callFake(function () {
-        return $q.resolve(ContractAPIMock.all());
+      /*
+       * @TODO the way Contact model is constructed now it is not possible to use
+       * the same syntax as used for other spies as the scope for Contact will be changed
+       * and "this.mockedContracts" will not work
+       */
+      spyOn(Contact, 'all').and.callFake(function () {
+        return $q.resolve(ContactAPIMock.all());
       });
+      spyOn(Contract, 'all').and.callFake(ContractAPIMock.all);
```

Comment:
You can just replace `ContractAPI` with `ContractAPIMock` and same goes for other `AbsenceType, AbsencePeriod  and Entitlement `. This way we dont need to spy on a specific method, instead we will be using the Mocked API itself.